scripts/manager_char_ca.lua

```diff
@@ -0,0 +1,66 @@
+--
+-- Please see the license file included with this distribution for
+-- attribution and copyright information.
+--
+
+
+local resetHealthOriginal;
+
+function onInit()
+	resetHealthOriginal = CharManager.resetHealth;
+	CharManager.resetHealth = resetHealth;
+end
+
+function resetHealth(nodeChar, bLong)
+    if bLong then
+        local rSource = ActorManager.resolveActor(nodeChar);
+        local tEffects = EffectManager5E.getEffectsByType(rSource, 'HDRECOVERY');
+        if next(tEffects) then
+            local nMod = 0;
+            local aClassHDInfo = {};
+            for _,tEffect in pairs(tEffects) do
+                if tEffect.mod > nMod then
+                    nMod = tEffect.mod;
+                    for _,vClass in ipairs(DB.getChildList(nodeChar, "classes")) do
+                        local aClassDice = DB.getValue(vClass, "hddie", {});
+                        if #aClassDice > 0 then
+                            aClassHDInfo[vClass] = tonumber(aClassDice[1]:sub(2)) or 0;
+                        end
+                    end
+                end
+            end
+            resetHealthOriginal(nodeChar, bLong);
```

Comment:
The original code should be invoked regardless of the bLong param. And related to that, it might be nice to have this functionality also interact with the the campaign's fast/slow recovery settings, though that is a bit tricky to do without reinventing the wheel given the current layout of the ruleset logic for that.